Button fix and scene transition

catdad666 opened the pull request.

whymog commented:
Nice job figuring out how to make this all work! Some notes for you:

At first I was confused why the `ButtonScript` game object exists as a child of `StartButton`. I can see what you were doing here, though, and it's a solution I wouldn't have considered! However, you could simplify it pretty substantially by moving `StartController.cs` off of the child and onto the parent, deleting the child, and dragging the script back onto your button logic. Make sure it's still deactivated when the scene runs and it should still work just fine.

However, this isn't a great pattern, either, because it adds additional potential for things to break. Plus, I'm 

You've already got a box collider on this parent game object that's aware when the mouse is inside or outside of it, so we can just add a script to `StartButton` telling it to listen for `OnMouseOver()` (more [here](https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnMouseOver.html)). For example, something like:

```c#
void OnMouseOver() {
    if (Input.GetMouseButtonDown(0)) {
        SceneManager.LoadScene("Room");
    }
}
```

Which is a lot like the script you had before!

Also, I like the green fade effect -- nice work! I'd recommend doing it a little differently on future projects, but I think this is fine for this game currently. (Canvases can be pretty resource-intensive, and you could've done the same effect with just a single sprite as a child of the main camera, probably. But it's not causing any problems.)

catdad666 commented:
Wow! This helped a lot. Figuring out the button took a good two and a half hours yesterday. I knew my eventual solution, which was a patchwork of what I found by searching, was way too complicated.

I implemented your much simpler solution. I also adjusted the fade in animation so it uses a sprite instead of a canvas and made the `FadeIn` into a prefab, just in case.

whymog commented:
Looks great! Nicely done 👍
